src/evm/protocol/vm/state_builder.rs

```diff
@@ -204,7 +204,12 @@ impl EVMPoolStateBuilder {
             self.id,
             self.tokens,
             self.block,
-            self.balances.unwrap_or_default(),
+            self.balances.ok_or_else(|| {
```

Comment:
Ok I wasn't sure about this in my PR. Let's move it to the 'new' function and make it not Option<> instead? 

That way it would be clearer from the interfaces. In general it's better to error at compile time than at runtime when possible.